pinot-query-planner/src/main/java/org/apache/calcite/sql/util/PinotChainedSqlOperatorTable.java

```diff
@@ -0,0 +1,91 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.calcite.sql.util;
+
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlOperatorTable;
+import org.apache.calcite.sql.SqlSyntax;
+import org.apache.calcite.sql.validate.SqlNameMatcher;
+
+import com.google.common.collect.ImmutableList;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * ============================================================================
+ * THIS CLASS IS COPIED FROM Calcite's {@link org.apache.calcite.sql.util.ChainedSqlOperatorTable} and modified the
+ * function lookup to terminate early once found from ordered SqlOperatorTable list.
+ * ============================================================================
+ *
+ * PinotChainedSqlOperatorTable implements the {@link SqlOperatorTable} interface by
+ * chaining together any number of underlying operator table instances.
+ */
+public class PinotChainedSqlOperatorTable implements SqlOperatorTable {
+  //~ Instance fields --------------------------------------------------------
+
+  protected final List<SqlOperatorTable> _tableList;
+
+  //~ Constructors -----------------------------------------------------------
+
+  public PinotChainedSqlOperatorTable(List<SqlOperatorTable> tableList) {
+    this(ImmutableList.copyOf(tableList));
+  }
+
+  /** Internal constructor; call {@link SqlOperatorTables#chain}. */
+  protected PinotChainedSqlOperatorTable(ImmutableList<SqlOperatorTable> tableList) {
+    _tableList = ImmutableList.copyOf(tableList);
+  }
+
+  //~ Methods ----------------------------------------------------------------
+
+  @Deprecated // to be removed before 2.0
+  public void add(SqlOperatorTable table) {
+    if (!_tableList.contains(table)) {
+      _tableList.add(table);
+    }
+  }
+
+  @Override public void lookupOperatorOverloads(SqlIdentifier opName,
+      @Nullable SqlFunctionCategory category, SqlSyntax syntax,
+      List<SqlOperator> operatorList, SqlNameMatcher nameMatcher) {
+    for (SqlOperatorTable table : _tableList) {
+      table.lookupOperatorOverloads(opName, category, syntax, operatorList,
+          nameMatcher);
+      // ======================================================================
+      // CHANGED LINES BELOW
+      // ======================================================================
+      if (!operatorList.isEmpty()) {
```

Comment:
Does this line mean as long as we find something, we stop looking further?

Reply:
if I understand calcite correctly (i spent a lot of time debugging this code) it'll just return everything from the first operator table - which might still have multiple matches within it.

@walterddr why is this necessary?

Reply:
this is b/c if a single operator returns a list of matches, it will NEVER result in multiple type of matches.

in our case b/c we remove the case sensitivity restriction. `FLOOR` is resolved into a `SqlFloorOperator` and a `UserDefinedSqlFunction("floor")` one from the std table and one from catalog. 

in this case calcite has some weird non-asserted casting that causes trouble.

Reply:
in fact i think this can be contribute back for sure.